packages/dart/lib/src/sentry_envelope.dart

```diff
@@ -96,6 +97,36 @@ class SentryEnvelope {
     );
   }
 
+  /// Create a [SentryEnvelope] containing raw log data payload.
+  /// This is used by the log batcher to send pre-encoded log batches.
+  @internal
+  factory SentryEnvelope.fromLogsData(
+    List<List<int>> encodedLogs,
+    SdkVersion sdkVersion,
+  ) {
+    // Create the payload in the format expected by Sentry
+    // Format: {"items": [log1, log2, ...]}
+    final payload = <int>[];
+    payload.addAll(utf8.encode('{"items":['));
+    for (int i = 0; i < encodedLogs.length; i++) {
+      if (i > 0) {
+        payload.addAll(utf8.encode(','));
+      }
+      payload.addAll(encodedLogs[i]);
+    }
+    payload.addAll(utf8.encode(']}'));
```

Comment:
we should use a `BytesBuilder(copy: false)` here

and then `builder.takeBytes()` similar to `FileSystemTransport`